client/src/Views/Home/Campaigns/Campaign/CampaignProspects.js

```diff
@@ -13,10 +13,62 @@ const useStyles = makeStyles({
 
 const CampaignProspects = props => {
   const classes = useStyles();
+  const [prospectsList, setProspectsList] = useState(props.prospects);
+  const [filteredProspects, setFilteredProspects] = useState(props.prospects);
+  const [recentlyFetched, setRecentlyFetched] = useState(false);
+
+  useEffect(() => {
+    if (props.prospects.length > 0 && !recentlyFetched) {
+      setProspectsList(props.prospects);
+      setFilteredProspects(props.prospects);
+      setRecentlyFetched(true);
+    }
+  });
```

Comment:
Be careful with having useEffects that don't have any dependency arrays - this will run every single time any state or props changes happen, which is rarely what you want to do

Reply:
Ah thanks, good info. Looks like if I set the props as the dependency array, it'll load correctly.